google-cloud-spanner/test/google/cloud/spanner/pool/keepalive_or_release_test.rb

```diff
@@ -1,126 +1,123 @@
+#   let(:pool) { client.instance_variable_get :@pool }
+#   let :results_hash do
+#     {
+#       metadata: {
+#         row_type: {
+#           fields: [
+#             { type: { code: :INT64 } }
+#           ]
+#         }
+#       },
+#       values: [ { string_value: "1" }]
+#     }
+#   end
+#   let(:results_grpc) { Google::Spanner::V1::PartialResultSet.new results_hash }
+#   let(:results_enum) { Array(results_grpc).to_enum }
+#
+#   before do
+#     # kill the background thread before starting the tests
+#     pool.instance_variable_get(:@keepalive_task).shutdown
+#   end
+#
+#   after do
+#     shutdown_client! client
+#   end
+#
+#   it "calls keepalive on the sessions that need it" do
+#     # update the session so it was last updated an hour ago
+#     session.instance_variable_set :@last_updated_at, Time.now - 60*60
+#     # set the session in the pool
+#     pool.all_sessions = [session]
+#     pool.session_queue = [session]
+#
+#     mock = Minitest::Mock.new
+#     mock.expect :execute_streaming_sql, results_enum, [session.path, "SELECT 1", transaction: nil, params: nil, param_types: nil, resume_token: nil, partition_token: nil, seqno: nil, options: default_options]
+#     session.service.mocked_service = mock
+#
+#     pool.keepalive_or_release!
+#
+#     shutdown_pool! pool
+#
+#     mock.verify
+#   end
+#
+#   it "calls keepalive on the transactions that need it" do
+#     # update the session so it was last updated an hour ago
+#     session.instance_variable_set :@last_updated_at, Time.now - 60*60
+#     # set the session in the pool
+#     pool.all_sessions = [session]
+#     pool.transaction_queue = [transaction]
+#
+#     mock = Minitest::Mock.new
+#     mock.expect :begin_transaction, transaction_grpc, [session_grpc.name, tx_opts, options: default_options]
+#     session.service.mocked_service = mock
+#
+#     pool.keepalive_or_release!
+#
+#     shutdown_pool! pool
+#
+#     mock.verify
+#   end
+#
+#   it "doesn't call keepalive on sessions that don't need it" do
+#     # update the session so it was last updated now
+#     session.instance_variable_set :@last_updated_at, Time.now
+#     # set the session in the pool
+#     pool.all_sessions = [session]
+#     pool.session_queue = [session]
+#
+#     mock = Minitest::Mock.new
+#     session.service.mocked_service = mock
+#
+#     pool.keepalive_or_release!
+#
+#     shutdown_pool! pool
+#
+#     mock.verify
+#   end
+#
+#   it "doesn't call keepalive on transactions that don't need it" do
+#     # update the session so it was last updated now
+#     session.instance_variable_set :@last_updated_at, Time.now
+#     # set the session in the pool
+#     pool.all_sessions = [session]
+#     pool.transaction_queue = [transaction]
+#
+#     mock = Minitest::Mock.new
+#     session.service.mocked_service = mock
+#
+#     pool.keepalive_or_release!
+#
+#     shutdown_pool! pool
+#
+#     mock.verify
+#   end
```

Comment:
Why is this file commented out?

Reply:
I was getting segfaults on Ruby 2.5, so I commented this test out to run some tests. I should have restored the file before committing. Thanks for pointing this out. Fixed now.